lib/components/isometric_tile_map_component.dart

```diff
@@ -0,0 +1,106 @@
+import 'dart:ui';
+
+import 'package:flame/components/component.dart';
+
+import '../flame.dart';
+import '../position.dart';
+import '../sprite.dart';
+
+class IsometricTileset {
+  final Image tileset;
+  final int size;
+
+  final Map<int, Sprite> _spriteCache = {};
+
+  IsometricTileset(this.tileset, this.size);
+
+  int get cols => tileset.width ~/ size;
+
+  Sprite getTile(int tileId) {
+    return _spriteCache[tileId] ??= _computeTile(tileId);
+  }
+
+  Sprite _computeTile(int tileId) {
+    final i = tileId % cols;
+    final j = tileId ~/ cols;
+    final s = size.toDouble();
+    return Sprite.fromImage(tileset, x: s * i, y: s * j, width: s, height: s);
+  }
+
+  static Future<IsometricTileset> load(String fileName, int size) async {
+    final image = await Flame.images.load(fileName);
+    return IsometricTileset(image, size);
+  }
+}
+
+class Block {
+  int x, y;
+  Block(this.x, this.y);
+
+  @override
+  String toString() => '($x, $y)';
+}
+
+class IsometricTileMapComponent extends PositionComponent {
+  IsometricTileset tileset;
+  List<List<int>> matrix;
+  int destTileSize;
+
+  IsometricTileMapComponent(this.tileset, this.matrix, {this.destTileSize});
+
+  int get effectiveTileSize => destTileSize ?? tileset.size;
+
+  @override
+  void render(Canvas c) {
+    prepareCanvas(c);
+
+    final size = Position.fromInts(effectiveTileSize, effectiveTileSize);
+    matrix.asMap().forEach((i, line) {
```

Comment:
Couldn't we just use two conventional `fors` here? I imagine that `asMap` could add some perfomance overhead

Reply:
I thought asMap only returned a view over the list. but I can change

Reply:
well actually I need the index. that is why I used the asMap

Reply:
that is why I am suggesting on using conventional `fors`

```dart
for (int y = 0; y < matrix.length; y++) {
...
```

Reply:
omg I forgot those existed AHAHAHAHA